### PR TITLE
chore: enable production settings only for merged ITs

### PR DESCRIPTION
**Pull request by sissbruecker**
## Description

Currently all Jetty configurations run with production settings by default. That is inconvenient for developers working with this repository as:
- production mode disables several useful features such as the dev mode tool, or the 404 fallback page listing available routes
- the frontend production bundle prevents from debugging JS code like connectors

This PR changes the setup to use development settings by default (no production mode, no production frontend bundle), and only enable those when running merged ITs. There might be other concerns that I'm not aware of, please address those in the review.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -84,7 +84,7 @@
     </modules>
     <profiles>
         <profile>
-            <id>default</id>
+            <id>merged-it</id>
```
Comment (sissbruecker) [inline pom.xml:87]: This profile does not seem to have any relation with the other `default` profiles declared in the individual integration test modules, as it uses a different activation. Seemed appropriate to rename it.

**Review by tomivirkki — APPROVED** [review]
Works great by default locally

File: pom.xml (hunk 1 above)
Comment (manolo) [reply pom.xml:87]: agree

Comment (manolo) [reply pom.xml:87]: agree

**Review by manolo — APPROVED** [review]

**Pull request merged**